PR for #93: Param to indicate file type

BenBlazely opened the pull request:
* Added support for specifing the type of a file. Use files: [{path: 'path/to/file.x', type: 'type'}] where type is either css, html or js. Co-exists with regular path-only files entries. Small 8 line change to address #93

ocombe commented:
Hey thanks @BenBlazely, can you also update the Readme & add a test (if you can, you don't have to but that'd be better) please ?

BenBlazely commented:
Travis and I aren't getting along this morning it seems...

BenBlazely commented:
Ok, fixed the module issue I introduced, but now Travis is complaining about not being able to find the new CSS file I added for testing. 

Error: Unable to load /base/tests/unit/lazyLoad/test.NoExtCss

The test.NoExtCss file is in the unit/lazlyLoad folder though... /confused

Apologies, Travis CI isn't something I'm very familiar with, could you take a look when you get a moment and let me know how I broke it? :-)

ocombe commented:
There was a mistake in the test, I fixed it and merged the PR in https://github.com/ocombe/ocLazyLoad/commit/725ceb9a51986b67a770de12b0cdd810e92cfb2d
Thanks again for all of this!

ocombe commented:
Ok I found out why it is not working on Travis, I run my local tests on Chrome but Travis uses Firefox.
With firefox when you try to load a css file that is not .css you will get the following error:
`The stylesheet [...]test.fakecss has not been loaded because its MIME type, « text/plain », is not « text/css »`

The server should serve the file with the good mime type. It's too complicated to do this in Travis so I changed the test to template injection instead :)

BenBlazely commented:
Silly Travis.

Thanks, it was driving me nuts.

ocombe commented:
It's not so much travis than firefox :)

BenBlazely commented:
Silly Firefox. 

(To be fair, Chrome on my Windows PC shows the same exception in the console, but it's only a warning and it still loads the sheet)
